apps/Gruntfile.js

```diff
@@ -926,7 +926,14 @@ describe('entry tests', () => {
           new TerserPlugin({
             // Excludes these from minification to avoid breaking functionality,
             // but still adds .min to the output filename suffix.
-            exclude: [/\/blockly.js$/, /\/brambleHost.js$/]
+            exclude: [/\/blockly.js$/, /\/brambleHost.js$/],
+            terserOptions: {
+              // Handle Safari 10.x issues: [See FND-2108 / FND-2109]
+              // Can remove when we can safely drop support for older iPad/iOS.
+              mangle: {
+                safari10: true
+              }
```

Comment:
Why do this as a configuration of `mangle` instead of
```
terserOptions: {
  safari10: true
}
```

I see `safari10` in https://webpack.js.org/plugins/terser-webpack-plugin/#terseroptions
and https://github.com/terser/terser#minify-options describes that it would do a little bit more than what you have here (if I'm understanding the docs correctly).

Reply:
Yeah. That's my understanding, too.

As far as I understand, there are (at least?) two `safari10` modes for different parts of the process. This one is just for name mangling, which is the exact issue at play here... it forces it to choose unique names in those contexts where Safari 10 mistakenly won't allow shadowing. The other is for an `await` bug interacting with loops or some such, which I don't think matters here. Probably doesn't hurt to use the global setting, which turns on all of the `safari10` modes... but then again, I don't know.